e2e/support/helpers/e2e-snowplow-helpers.js

```diff
@@ -28,15 +28,27 @@ export const blockSnowplow = () => {
  * @param {number} count - number of matching events you expect to find. defaults to 1
  */
 export const expectGoodSnowplowEvent = (eventData, count = 1) => {
+  let lastReceivedEvent = null;
   retrySnowplowRequest(
     "micro/good",
-    ({ body }) =>
-      body.filter(snowplowEvent =>
-        isDeepMatch(
-          snowplowEvent?.event?.unstruct_event?.data?.data,
-          eventData,
-        ),
-      ).length === count,
+    ({ body }) => {
+      lastReceivedEvent = body?.[0].event?.unstruct_event?.data?.data;
+
+      return (
+        body.filter(snowplowEvent =>
+          isDeepMatch(
+            snowplowEvent?.event?.unstruct_event?.data?.data,
+            eventData,
+          ),
+        ).length === count
+      );
+    },
+    () =>
```

Comment:
needs to be a function otherwise the string will be created when lastReceivedEvent is still `null`